pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -478,20 +486,35 @@ public synchronized void buildRouting(String tableNameWithType) {
         Set<String> offlineTablePreSelectedOnlineSegments =
             offlineTableSegmentPreSelector.preSelect(offlineTableOnlineSegments);
         TimeBoundaryManager offlineTableTimeBoundaryManager =
-            new TimeBoundaryManager(offlineTableConfig, _propertyStore, _brokerMetrics);
-        offlineTableTimeBoundaryManager.init(offlineTableIdealState, offlineTableExternalView,
-            offlineTablePreSelectedOnlineSegments);
-        offlineTableRoutingEntry.setTimeBoundaryManager(offlineTableTimeBoundaryManager);
+            new TimeBoundaryManager(offlineTableConfig, schema, _brokerMetrics);
+        offlineTableRoutingEntry.initTimeBoundaryManager(offlineTableTimeBoundaryManager, offlineTableIdealState,
+            offlineTableExternalView, offlineTablePreSelectedOnlineSegments);
```

Comment:
This init is not protected by synchronize and thus I cannot move it to be init via SegmentZkMetadataCache...

This potentially can cause out-of-sync between the cache and the TimeBoundaryManager

In order for this to be called via SegmentZkMetadataCache, we can either:
- make onAssignmentChange API call (which is synchronized);
- make the init method also synchronized

Reply:
chatted offline with @Jackie-Jiang ^ we decided to leave timboundary out of the segment zk metadata fetcher. 
1. timeboundary manager is init separately and can have different timing than the rest of the zk required utilities. 
2. it is very possible that during server start up no zk metadata fetcher is needed (e.g. no pruner and others).
    - enabling this by default on all table is a waste
    - merging this with the rest will possible cause duplicate init and slow down server start up

will comment on the javadoc to make sure this knowledge is documented after the PR is merged